src/lib/core/CHIPTLVReader.cpp

```diff
@@ -207,35 +207,45 @@ CHIP_ERROR TLVReader::Get(uint64_t & v)
     return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR TLVReader::Get(double & v)
+namespace {
+template <typename T>
+CHIP_ERROR GetFloatingImpl(const TLVElementType elementType, const uint64_t elemLenOrVal, T & v)
 {
-    switch (ElementType())
+    switch (elementType)
     {
     case TLVElementType::FloatingPointNumber32: {
-        union
-        {
-            uint32_t u32;
-            float f;
-        } cvt;
-        cvt.u32 = static_cast<uint32_t>(mElemLenOrVal);
-        v       = cvt.f;
+        float f;
+        auto u32 = static_cast<uint32_t>(elemLenOrVal);
+        memcpy(&f, &u32, sizeof(f));
```

Comment:
Should probably do a followup for a similar fix to CHIPTLVWriter?

Reply:
I will send a follow-up PR for this.

Reply:
#8688

Reply:
Apologies, my brain was thinking in terms of Gerrit and not pull requests. Added to this PR as a separate commit.